src/components/DatasetPreview.vue

```diff
@@ -1,26 +1,80 @@
 <template>
     <div class="py-3">
         <h3>{{ dataset.name }}</h3>
-        <span v-html="this.description"></span>
+        <v-chip v-for="term in includedTerms" 
+                :key="term.termUri" @click="handleChipClick(term.termName)" 
+                small :color="getChipColor(term.objectClass)">
+                {{ term.termName }} 
+          </v-chip> 
+        <div v-html="this.description"></div>
     </div>
 </template>
 
 <script>
 import { highlight } from "@/search-utils";
-import { marked } from "@/config/gemma";
+import { marked, axiosInst, baseUrl, excludedTerms, excludedCategories } from "@/config/gemma";
 
 export default {
   name: "DatasetPreview",
   props: {
-    dataset: Object
+    dataset: Object    
+  },
+  data() {
+    return {
+      terms: [],
+      includedTerms: []
+    }
   },
   computed: {
     description() {
       if (this.dataset.searchResult !== undefined && this.dataset.searchResult.highlights !== null && "description" in this.dataset.searchResult.highlights) {
         return marked.parseInline(highlight(this.dataset.description, this.dataset.searchResult.highlights.description));
       }
       return marked.parseInline(this.dataset.description);
+    },
+    chipColorMap() {
+      return {
+        FactorValue: 'yellow',
+        ExperimentTag: 'green',
+        BioMaterial: 'blue'
+      };
     }
+  },
+  methods: {
+    getTerms() {
+      const dataset = this.dataset.id
+      return axiosInst.request({
+        method: 'GET',
+          url: baseUrl + `/rest/v2/datasets/${dataset}/annotations`,
+          params: {
+            filter: `id = ${dataset}`
+          }
```

Comment:
The `filter` argument only applies to the `/rest/v2/datasets/annotations` endpoint.